charts/featureform/templates/api-server/rollout.yaml

```diff
@@ -0,0 +1,27 @@
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+#  Copyright 2024 FeatureForm Inc.
+#
+
+{{ if .Values.argo.rollouts.enabled }}
+apiVersion: argoproj.io/v1alpha1
+kind: Rollout
+metadata:
+  name: featureform-api-server
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: featureform-api-server
+  workloadRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: featureform-api-server
+    scaleDown: onsuccess
+  strategy:
+    blueGreen:
+      activeService: featureform-api-server
+      autoPromotionEnabled: false
```

Comment:
q: `autoPromotionEnabled`  set to `false` keeps the actual flip to the new version manual right?

Reply:
That's correct, in this case there's a manual step before switching over to the green deployment, as outlined here https://argo-rollouts.readthedocs.io/en/stable/generated/kubectl-argo-rollouts/kubectl-argo-rollouts_promote/. Figured it would be a safer default to allow customers to switch at whatever point they're comfortable.

Reply:
I agree that's probably the better approach to avoid hiccups between the 2 versions.

Reply:
resolve at will.